### PR TITLE
refactor(ingest): streamline pydantic configs

### PR DESCRIPTION
**Pull request by hsheth2**
## Checklist
- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: metadata-ingestion/src/datahub/configuration/validate_field_rename.py [hunk 1]
```diff
@@ -22,10 +23,12 @@ def _validate_field_rename(cls: Type, values: dict) -> dict:
                     f"Cannot specify both {old_name} and {new_name} in the same config. Note that {old_name} has been deprecated in favor of {new_name}."
                 )
             else:
-                warnings.warn(
-                    f"The {old_name} is deprecated, please use {new_name} instead.",
-                    UserWarning,
-                )
+                if print_warning:
+                    warnings.warn(
```
Comment (rslanka) [inline metadata-ingestion/src/datahub/configuration/validate_field_rename.py:27]: Do we already set logging.captureWarnings(TRUE) in our logging settings? See [this](https://docs.python.org/3/library/logging.html#logging.captureWarnings).

File: metadata-ingestion/src/datahub/ingestion/source/file.py [hunk 2]
```diff
@@ -179,16 +173,15 @@ def create(cls, config_dict, ctx):
         return cls(ctx, config)
 
     def get_filenames(self) -> Iterable[str]:
-        is_file = os.path.isfile(self.config.path)
-        is_dir = os.path.isdir(self.config.path)
+        is_file = self.config.path.is_file()
+        is_dir = self.config.path.is_dir()
```
Comment (rslanka) [inline metadata-ingestion/src/datahub/ingestion/source/file.py:177]: Get rid of `is_file` and `is_dir` vars?

File: metadata-ingestion/src/datahub/configuration/validate_field_rename.py (hunk 1 above)
Comment (hsheth2) [reply metadata-ingestion/src/datahub/configuration/validate_field_rename.py:27]: We don't, and I don't think we want to - for the end user, the message gets printed out in a clearer way with warnings. Also, not capturing lets us make assertions based on warnings in tests

File: metadata-ingestion/src/datahub/ingestion/source/file.py (hunk 2 above)
Comment (hsheth2) [reply metadata-ingestion/src/datahub/ingestion/source/file.py]: fixed!

File: metadata-ingestion/src/datahub/configuration/validate_field_rename.py (hunk 1 above)
Comment (rslanka) [reply metadata-ingestion/src/datahub/configuration/validate_field_rename.py:27]: How about UI based ingestion?

**Review by shirshanka — APPROVED** [review]
LGTM

**Pull request merged**